packages/sdk-cli/src/models/DeveloperRelay.test.ts

```diff
@@ -0,0 +1,186 @@
+import nock from 'nock';
+import stream from 'stream';
+import websocketStream from 'websocket-stream';
+
+import * as auth from '../auth';
+import environment from '../auth/environment';
+import DeveloperRelay from './DeveloperRelay';
+import mockWithPromiseWaiter from '../testUtils/mockWithPromiseWaiter';
+
+jest.mock('websocket-stream', () => jest.fn());
+jest.mock('../auth');
+
+const mockHostID = 'fakeHost';
+
+const mockAppHost = {
+  id: 'apphostA',
+  displayName: 'App Host',
+  roles: ['APP_HOST'],
+  state: 'available',
+};
+
+const mockCompanionHost = {
+  id: 'companionhostA',
+  displayName: 'Companion Host',
+  roles: ['COMPANION_HOST'],
+  state: 'available',
+};
+
+let endpointMock: nock.Scope;
+
+function mockHostsResponse(status: number, payload?: any) {
+  endpointMock
+    .get('/1/user/-/developer-relay/hosts.json')
+    .reply(status, payload);
+}
+
+function mockHostsSuccessResponse(hosts = [mockAppHost, mockCompanionHost]) {
+  mockHostsResponse(200, { hosts });
+}
+
+function mockConnectionURLResponse(hostID: string, response: string) {
+  endpointMock
+    .post(`/1/user/-/developer-relay/hosts/${hostID}`)
+    .reply(200, response, { 'Content-Type': 'text/uri-list' });
+}
+
+describe.each([
+  {
+    shouldAuth: true,
+    authToken: 'mockToken',
+    apiUrl: environment().config.apiUrl,
+    describeMessage: 'default: require auth, valid auth token',
+  },
+  {
+    // If auth was to run, it would fail (because authToken = null).
+    // This is used to test that auth configuration is properly passed to apiFetch and auth DOESN'T run.
+    shouldAuth: false,
+    authToken: null,
+    apiUrl: 'https://fake-dev-relay-test-endpoint.fitbit.com',
+    describeMessage:
+      "configurable: don't require auth, invalid auth token, dummy API URL",
+  },
+])('$describeMessage', ({ shouldAuth, authToken, apiUrl }) => {
+  let developerRelay: DeveloperRelay;
+
+  // Custom API endpoint
+  beforeAll(() => {
+    developerRelay = new DeveloperRelay(apiUrl, shouldAuth);
+  });
+
+  beforeEach(() => {
+    (auth.getAccessToken as jest.Mock).mockResolvedValueOnce(authToken);
+    endpointMock = nock(apiUrl);
+  });
+
+  afterAll(() => {
```

Comment:
Where are these mocked manually in order to need to be manually restored after all tests?